akka-cluster-sharding/src/multi-jvm/scala/akka/cluster/sharding/MultiNodeClusterShardingSpec.scala

```diff
@@ -72,67 +105,119 @@ abstract class MultiNodeClusterShardingSpec(val config: MultiNodeClusterSharding
     super.afterTermination()
   }
 
-  protected def join(from: RoleName, to: RoleName): Unit = {
+  /**
+   * Flexible cluster join pattern usage.
+   *
+   * @param from the node the `Cluster.join` is `runOn`
+   * @param to the node to join to
+   * @param onJoinedRunOnFrom optionally execute a function after join validation
+   *                          is successful, e.g. start sharding or create coordinator
+   * @param assertEnabled if disabled - false, the joining member's `MemberStatus.Up`
+   *                      and similar assertions are not run. This allows tests that were
+   *                      not doing assertions (e.g. ClusterShardingMinMembersSpec) or
+   *                      doing them after `onJoinedRunOnFrom` more flexibility.
+   *                      Defaults to true, running member status checks.
+   */
+  protected def join(
+      from: RoleName,
+      to: RoleName,
+      onJoinedRunOnFrom: => Unit = (),
+      assertEnabled: Boolean = true,
```

Comment:
`assertNodeUp` or something else saying what it is actually asserting?

Reply:
Done